examples/typescript-reactflux/src/stores/TodoStore.ts

```diff
@@ -0,0 +1,196 @@
+}
+
+/**
+ * Update all of the TODO items with the same object.
+ *     the data to be updated.  Used to mark all TODOs as completed.
+ * @param  {object} updates An object literal containing only the data to be
+ *     updated.
+
+ */
+function updateAll(updates:TodoUpdateData): void {
+  var id: string;
+  for (id in _todos) {
+    if(_todos.hasOwnProperty(id)) {
+      update(id, updates);
+    }
+  }
+}
+
+/**
+ * Delete a TODO item.
+ * @param  {string} id
+ */
+function destroy(id: string): void {
+  delete _todos[id];
+}
+
+/**
+ * Delete all the completed TODO items.
+ */
+function destroyCompleted(): void {
+  var id: string;
+  for (id in _todos) {
+    if (_todos[id].complete) {
+      destroy(id);
+    }
+  }
+}
+
+class TodoStoreStatic extends events.EventEmitter {
+
+  /**
+   * Tests whether all the remaining TODO items are marked as completed.
+   * @return {boolean}
+   */
+  public areAllComplete(): boolean {
+    var id: string;
+
+    for(id in _todos) {
+      if(!_todos[id].complete) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Get the entire collection of TODOs.
+   * @return {object}
+   */
+  public getAll(): TodoMap {
+    return _todos;
+  }
+
+  public emitChange(): void {
+    this.emit(CHANGE_EVENT);
+  }
+
+  /**
+   * @param {function} callback
+   */
+  public addChangeListener(callback: () => void): void {
+    this.on(CHANGE_EVENT, callback);
+  }
+
+  /**
+   * @param {function} callback
+   */
+  public removeChangeListener(callback: () => void) {
+    this.removeListener(CHANGE_EVENT, callback);
+  }
+}
+
+var TodoStore: TodoStoreStatic =  new TodoStoreStatic();
+
+// Register callback to handle all updates
+AppDispatcher.register( function(action:TodoAction): void {
+  var text: string;
+
+  switch(action.actionType) {
+    case TodoConstants.TODO_CREATE:
+      text = action.text.trim();
+      if (text !== '') {
+        create(text);
+      }
+      TodoStore.emitChange();
+      break;
+
+    case TodoConstants.TODO_TOGGLE_COMPLETE_ALL:
+      if (TodoStore.areAllComplete()) {
+        updateAll({complete: false});
+      } else {
+        updateAll({complete: true});
+      }
+      TodoStore.emitChange();
+      break;
+
+    case TodoConstants.TODO_UNDO_COMPLETE:
+      update(action.id, {complete: false});
+      TodoStore.emitChange();
+      break;
+
+    case TodoConstants.TODO_COMPLETE:
+      update(action.id, {complete: true});
+      TodoStore.emitChange();
+      break;
+
+    case TodoConstants.TODO_UPDATE_TEXT:
+      text = action.text.trim();
+      if (text !== '') {
+        update(action.id, {text: text});
+      }
+      TodoStore.emitChange();
+      break;
+
+    case TodoConstants.TODO_DESTROY:
+      destroy(action.id);
+      TodoStore.emitChange();
+      break;
+
+    case TodoConstants.TODO_DESTROY_COMPLETED:
+      destroyCompleted();
+      TodoStore.emitChange();
+      break;
+
+    default:
+      // no op
```

Comment:
> I would like to avoid changing Facebook code.

You will need to anyways. By the way "facebook code", is this takeover from an example they built? If so I'd love to make sure they know about this, and more importantly, that they agree on including it. Not only talking about the legal side of things but also it's just friendly to let them know and ask what they think.

Besides from that, the app is unlikely to be spec compliant so you'll need to change it anyways. 

About this line in specific, there's, really, totally no deal with including this.